src/app/loadout/loadout-ui/menu-hooks.tsx

```diff
@@ -225,19 +207,45 @@ export function searchAndSortLoadoutsByQuery(
   language: DimLanguage,
   loadoutSort: LoadoutSort
 ) {
-  const loadoutQueryPlain = plainString(query, language);
-  return _.sortBy(
-    loadouts.filter(
-      (loadout) =>
-        !query ||
-        plainString(loadout.name, language).includes(loadoutQueryPlain) ||
-        (!isInGameLoadout(loadout) &&
-          loadout.notes &&
-          plainString(loadout.notes, language).includes(loadoutQueryPlain))
-    ),
-    (l) => (isInGameLoadout(l) ? 0 : 1),
+  const sortCollator = new Intl.Collator(language, {
+    numeric: true,
+    usage: 'sort',
+    sensitivity: 'accent',
+  });
+  const filterCollator = new Intl.Collator(language, { usage: 'search', sensitivity: 'base' });
```

Comment:
These crash with our `zh-cht` and `zh-chs` languages because browsers expect either `zh` or something like `zh-Hant` and `zh-Hans`.